### DISCUSSION TITLE
Writing a wel package in a modflow model doesn't work properly

### DISCUSSION THREAD

**Discussion by YANG-fq** [modflowpy/flopy#2224] [Q&A] [answered]
Hi,
I created the well package using flopy.modflow.ModflowWel after importing the modflow model from gms in flopy, and then ran the modflow model. No error is prompted, but just can't run the model properly to get the simulation result.Could you please help me in this regard?
Thanks in advance for your answers.
Here is my simple code:
![code](https://github.com/modflowpy/flopy/assets/172515995/c8e4837c-a27c-4e30-b803-b6aeb51f1f29)

And here is the error:
![Uploading result.jpg…]()

**Comment by wpbonelli** [comment]
We are not able to see the error message @YANG-814

**Reply by YANG-fq** [reply]
![result](https://github.com/modflowpy/flopy/assets/172515995/878e6d2a-8964-4640-8cd2-0b4532a43434)

**Reply by wpbonelli** [reply]
is there an error message from mf2005 or does it seem to terminate successfully (i.e. flopy misreporting the status?)

**Reply by YANG-fq** [reply]
Here's my run, if I comment out the wel package I can get the normal run model.Bro, do you know what is the reason for this? @wpbonelli

**Reply by YANG-fq** [reply]
> is there an error message from mf2005 or does it seem to terminate successfully (i.e. flopy misreporting the status?)

terminate successfully

**Reply by wpbonelli** [reply]
hard to say what the issue is without seeing your model, can you upload the full script or full set of input files?

**Reply by YANG-fq** [reply]
> > is there an error message from mf2005 or does it seem to terminate successfully (i.e. flopy misreporting the status?)
> 
> terminate successfully

like this
![result](https://github.com/modflowpy/flopy/assets/172515995/804e5c8d-9c6f-4e50-98b1-2005678da3df)

**Reply by wpbonelli** [reply]
we will also need the input files for the model you are loading and modifying

**Reply by jlarsen-usgs** [reply, marked as answer]
Have you opened your modflow list file and looked to see if it contains any error messages (e.g., model did not converge, input errors, etc...)

I'd also note that the `stress_period_data` is usually provided in a dictionary to boundary condition packages, here's an example:

```python
nper = mf.dis.nper
pumping = [
[76 - 1,99 - 1,1 - 1,-10],
[81 - 1,98 - 1,1 - 1,-10],
[85 - 1,102 - 1,1 - 1,-10],
]
wel_spd = {i: pumping for i in range(nper)}
wel = flopy.modflow.ModflowWel(mf, stress_period_data=wel_spd)
mf.write_input()
```

Your pumping record also looks odd to me. Do you have 85 or more layers in your model?

**Reply by YANG-fq** [reply]
> Your pumping record also looks odd to me. Do you have 85 or more layers in your model?

Thank you very much, I had reversed the position of the layers and columns so that caused my model to be incorrect. Thank you so much for solving a problem that has been bothering me all day, good luck with your work and have a great day!

**Reply by YANG-fq** [reply]
Thank you very much for answering my question, I've found the error, I've reversed the position of the layers and columns and that's causing the model error. I'm very sorry for taking up your time due to my carelessness!
Good luck with your work and have a great day! @wpbonelli
